app/services/post_alerter.rb

```diff
@@ -265,6 +269,18 @@ def category_watchers(topic)
       .pluck(:user_id)
   end
 
+  def category_or_tag_muters(topic)
+    User
+      .joins(
+        "LEFT JOIN category_users ON users.id = category_users.user_id AND category_users.category_id = #{topic.category_id.to_i} AND category_users.notification_level = #{CategoryUser.notification_levels[:muted].to_i}",
+      )
+      .joins("LEFT JOIN topic_tags ON topic_tags.topic_id = #{topic.id.to_i}")
+      .joins(
+        "LEFT JOIN tag_users ON users.id = tag_users.user_id AND tag_users.tag_id IN (tag_users.tag_id) AND tag_users.notification_level = #{TagUser.notification_levels[:muted].to_i}",
```

Comment:
`tag_users.tag_id IN (tag_users.tag_id)`

This keeps regressing, is there a test case we're missing here?

Reply:
yeah, it was always returning topics to mute 🙈 Added spec which is failing with wrong join